### ISSUE TITLE
Remove method `count.tbl_sql()`

### ISSUE THREAD

**Issue by mgirlich** [tidyverse/dplyr#6338]
dplyr defines a method `count.tbl_sql()`. dbplyr defines a method `count.tbl_lazy()`. Now, the dplyr method wins due to the classes of database tables. I guess it would make sense to simply remove the method from dplyr (same issue for `tally.tbl_sql()`).

**Comment by DavisVaughan** [comment]
Oh but its `count.tbl_lazy` in dbplyr? I guess that was a change

**Comment by DavisVaughan** [comment]
It looks like `count()` was made generic here, which is also when the `tbl_sql` method was added:
https://github.com/tidyverse/dplyr/pull/5633

Making `count()` generic seems to be a direct response from a request to do so so it can be used in dbplyr here
https://github.com/tidyverse/dbplyr/commit/4dbcb44ea90b1a8dd4221b84c06bbf95e831cdbd

The original dplyr PR was using `tbl_sql`, and the original dbplyr PR was using `tbl_lazy`, so I'm guessing it has been wrong this whole time? Romain even says in this comment that he didn't test it "with a dbplyr that has the methods" so he wouldn't have seen that it was broken https://github.com/tidyverse/dplyr/pull/5633#issuecomment-738848253

I think we could try to:
- Remove the methods
- Remove the onLoad conditional registration
- Remove the `if (is.data.frame(x))` check from `count.data.frame()` and always unconditionally reconstruct. This seems to be a "hack" that was added to support data bases, added in https://github.com/tidyverse/dplyr/pull/5114/, which wouldn't be needed if we removed these methods. It looks like that hack was added to `add_count()` then too, but has already been removed from there since then.
